### PR TITLE
Migrate to ESM, ESLint 9 and flat config

### PR DESCRIPTION
**Pull request by philippfromme**
Updates the library to support eslint@9 (only), as eslint@8 is out of maintainance, as of October, 1st.

Turns this into an ESM only package, so consume it in `.mjs` files in commonjs projects.

----

Related to https://github.com/bpmn-io/internal-docs/issues/1042
Related to https://github.com/bpmn-io/internal-docs/issues/883

### CODE REVIEW

File: configs/commercial.js [hunk 1]
```diff
@@ -1,10 +1,14 @@
-const path = require('path');
+import licenseHeaderPlugin from 'eslint-plugin-license-header';
 
-module.exports = {
-  plugins: [
-    'license-header'
-  ],
-  rules: {
-    'license-header/header': [2, path.join(__dirname, '/../resources/commercial-license-header.js') ]
+import commercialLicenseHeader from '../resources/commercial-license-header.js'
```
Comment (nikku) [inline configs/commercial.js]: Retrieve a file path, local from the current file, in an ES module:

```
const { fileURLToPath } = require('node:url');

const path = fileURLToPath(new URL('../somePath/relative.path', import.meta.url));
```

Cf. [URL](https://nodejs.org/api/url.html) for tricks available.

**Review by nikku — CHANGES_REQUESTED** [review]

File: configs/commercial.js (hunk 1 above)
Comment (philippfromme) [reply configs/commercial.js]: ✅

File: configs/commercial.js [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
 
-module.exports = {
-  plugins: [
-    'license-header'
-  ],
-  rules: {
-    'license-header/header': [2, path.join(__dirname, '/../resources/commercial-license-header.js') ]
+import { fileURLToPath } from 'node:url';
+
+const path = fileURLToPath(new URL('../resources/commercial-license-header.txt', import.meta.url));
```
Comment (nikku) [inline configs/commercial.js]: I'm not sure why this is a `.txt` file, as it is (must be) a valid JavaScript snippet.

**Review by nikku — COMMENTED** [review]
Looks good from my end. One thing to consider: https://github.com/camunda/eslint-plugin-camunda-licensed/pull/13/files#r1791441298.

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -6,8 +6,8 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [ 14, 16 ]
-        eslint: [ 7, '' ]
+        node-version: [ 18, 20, 22 ]
+        eslint: [ 9, '' ]
```
Comment (nikku) [inline .github/workflows/CI.yml]: Empty = latest version.

```suggestion
        eslint: [ '' ]
```

File: configs/commercial.js (hunk 2 above)
Comment (philippfromme) [reply configs/commercial.js]: Done.

File: package.json [hunk 4]
```diff
@@ -2,7 +2,13 @@
   "name": "eslint-plugin-camunda-licensed",
   "version": "0.4.6",
   "description": "Shared lint rules for Camunda licensed OSS projects",
-  "main": "index.js",
+  "type": "module",
+  "exports": {
+    ".": "./index.js",
+    "./configs/apache.js": "./configs/apache.js",
+    "./configs/commercial.js": "./configs/commercial.js",
+    "./configs/mit.js": "./configs/mit.js"
```
Comment (nikku) [inline package.json]:
```suggestion
    "./configs/mit.js": "./configs/mit.js",
    "./package.json": "./package.json"
```

**Review by nikku — CHANGES_REQUESTED** [review]
Two small nit-picks, and potential to include https://github.com/bpmn-io/eslint-plugin-bpmn-io/commit/55c414dc6e7deeea717115487b65439f886afe3e.

File: package.json (hunk 4 above)
Comment (nikku) [inline package.json]:
```suggestion
    "./configs/*.js": "./configs/*.js"
```

Cf. https://github.com/bpmn-io/eslint-plugin-bpmn-io/blob/main/package.json#L8.

**Review by nikku — APPROVED** [review]

**Comment by nikku** [comment]
Thanks! I'll go ahead and merge + release this.

**Pull request merged**